### DISCUSSION TITLE
brew bump-revision fails to gpg sign commit

### DISCUSSION THREAD

**Discussion by ipatch** [Homebrew/discussions#3527] [Tap maintenance and Homebrew development] [answered]
i use gpg to sign my commit, and i'm running `brew bump-revision --message "rebuild for mojave" freecad/freecad/pyside2@5.15.5` and am getting the below error message,

```
error: gpg failed to sign the data
fatal: failed to write commit object
Error: Failure while executing; `git commit --no-edit --verbose --message=pyside2@5.15.5:\ revision\ bump\ rebuild\ pyside\ for\ mojave -- /Users/chris/homebrew/Library/Taps/freecad/homebrew-freecad/Formula/pyside2@5.15.5.rb` exited with 128.
```

typically when i run,

```
git add path/to/file
git commit -m "my commit msg"
```

i'm presented with a cli dialogue to sign my commit message but running `brew bump-revision` just shows the above error message. it however does edit the formula file.

**Comment by SMillerDev** [comment]
You'll need to debug your gpg setup. Commit signing isn't really something Homebrew forums can help you with.

**Comment by carlocab** [comment, marked as answer]
You'll need to find a way to cache your credentials, since `brew` doesn't handle in-terminal interaction well.

Alternatively, you could maybe try using `pinentry-mac` to avoid prompting for your credentials inside the terminal. It'll prompt you through a GUI dialogue instead.
